astroquery/query.py

```diff
@@ -421,20 +418,16 @@ def _download_file(self, url, local_filepath, timeout=None, auth=None,
         else:
             progress_stream = io.StringIO()
 
-        with ProgressBarOrSpinner(
-                length, ('Downloading URL {0} to {1} ...'
-                         .format(url, local_filepath)),
-                file=progress_stream) as pb:
+        with ProgressBarOrSpinner(length, f'Downloading URL {url} to {local_filepath} ...',
+                                  file=progress_stream) as pb:
             with open(local_filepath, open_mode) as f:
                 for block in response.iter_content(blocksize):
                     f.write(block)
-                    bytes_read += blocksize
+                    bytes_read += len(block)
```

Comment:
any chance you can explain why `len(block) != blocksize`?  Is one in bits and one in bytes or something?

Reply:
No, I suppose the factor of 8 comes from bytes in blocksize but then it becomes an int64? I would need to double-check, but in either case, this is also the approach in the original code.

Reply:
I don't grok the internals but `blocksize` seems to be from your config setting, so it is what you desire, while `len(block)` is the reality. And perhaps https://docs.python-requests.org/en/master/api/#requests.Response.iter_content explains it:

_Iterates over the response data. When stream=True is set on the request, this avoids reading the content at once into memory for large responses. The chunk size is the number of bytes it should read into memory. This is not necessarily the length of each item returned as decoding can take place._

Reply:
"This is not necessarily the length of each item returned as decoding can take place."  Ah.  I even read this yesterday but didn't put 2 and 2 together.